tests/repo_finder/test_repo_finder_deps_dev.py

```diff
@@ -0,0 +1,192 @@
+
+    if server_url:
+        target_url = f"/{deps_dev_service_mock['api']}/{deps_dev_service_mock['purl']}/pkg:{purl.type}/{purl.name}"
+        httpserver.expect_request(target_url).respond_with_data(data)
+
+    result, outcome = DepsDevRepoFinder().get_latest_version(purl)
+
+    assert not result
+    assert outcome == expected_outcome
+
+
+def test_get_latest_version_success(httpserver: HTTPServer, deps_dev_service_mock: dict) -> None:
+    """Test get latest version success."""
+    purl = PackageURL.from_string("pkg:pypi/test@3")
+    target_url = f"/{deps_dev_service_mock['api']}/{deps_dev_service_mock['purl']}/pkg:{purl.type}/{purl.name}"
+    httpserver.expect_request(target_url).respond_with_data(
+        '{"version": [{"versionKey":{"version": "4"}, "isDefault":true}]}'
+    )
+    result, outcome = DepsDevRepoFinder().get_latest_version(purl)
+    assert result
+    assert outcome == RepoFinderInfo.FOUND_FROM_LATEST
+
+
+@pytest.mark.parametrize(
+    ("purl_string", "server_url", "data"),
+    [
+        ("pkg:pypi/test", False, ""),
+        ("pkg:pypi/test@3", False, ""),
+        ("pkg:pypi/test@3", True, '{"foo": "bar"}'),
+        ("pkg:pypi/test@3", True, '{"attestations": [1, 2]}'),
+        ("pkg:pypi/test@3", True, '{"attestations": [{"url": "*replace_url*/bad_endpoint"}]}'),
+        ("pkg:pypi/test@3", True, '{"attestations": [{"url": "*replace_url*"}]}'),
+        ("pkg:pypi/test@3", True, '{"attestations": [{"url": "*replace_url*"}], "attestation_bundles": [1,2]}'),
+        (
+            "pkg:pypi/test@3",
+            True,
+            '{"attestations": [{"url": "*replace_url*"}], "attestation_bundles": [{"attestations": [1]}]}',
+        ),
+    ],
+)
+def test_get_attestation_failures(
+    httpserver: HTTPServer, deps_dev_service_mock: dict, purl_string: str, server_url: bool, data: str
+) -> None:
+    """Test get attestation failures."""
+    purl = PackageURL.from_string(purl_string)
+
+    if server_url:
+        assert purl.version
+        target_url = (
+            f"/{deps_dev_service_mock['api']}/"
+            + f"{'/'.join(['systems', purl.type, 'packages', purl.name, 'versions', purl.version])}"
+        )
+        if "*replace_url*" in data:
+            attestation_url = (
+                f"{deps_dev_service_mock['base_scheme']}://{deps_dev_service_mock['base_netloc']}{target_url}"
+            )
+            data = data.replace("*replace_url*", attestation_url)
+
+        httpserver.expect_request(target_url).respond_with_data(data)
+
+    result, _ = DepsDevRepoFinder().get_attestation(purl)
+    assert not result
+
+
+def test_get_attestation_success(httpserver: HTTPServer, deps_dev_service_mock: dict) -> None:
+    """Test get attestation success."""
+    purl = PackageURL.from_string("pkg:pypi/test@3")
+    target_url = (
+        f"/{deps_dev_service_mock['api']}/"
+        + f"{'/'.join(['systems', purl.type, 'packages', purl.name, 'versions', purl.version or ''])}"
```

Comment:
I wonder why we have `purl.version or ''` here. I think `purl` is not modified at all in this test.

Reply:
This has been removed as part of using PURL for attestation instead of the BASE endpoint.